app/uk/gov/hmrc/customs/notification/services/ClientWorkerImpl.scala

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.customs.notification.services
+
+import java.util.concurrent.TimeUnit
+import javax.inject.Singleton
+
+import akka.actor.ActorSystem
+import uk.gov.hmrc.customs.notification.connectors.{ApiSubscriptionFieldsConnector, NotificationQueueConnector, PublicNotificationServiceConnector}
+import uk.gov.hmrc.customs.notification.domain._
+import uk.gov.hmrc.customs.notification.logging.NotificationLogger
+import uk.gov.hmrc.customs.notification.repo.{ClientNotificationRepo, LockOwnerId, LockRepo}
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ClientWorkerImpl(
+                        config: CustomsNotificationConfig,
+                        actorSystem: ActorSystem,
+                        repo: ClientNotificationRepo,
+                        callbackDetailsConnector: ApiSubscriptionFieldsConnector,
+                        pushConnector: PublicNotificationServiceConnector,
+                        pullConnector: NotificationQueueConnector,
+                        lockRepo: LockRepo,
+                        logger: NotificationLogger
+                      ) extends ClientWorker {
+
+  private val extendLockDuration =  org.joda.time.Duration.millis(config.pushNotificationConfig.lockRefreshDurationInMilliseconds)
+  private val refreshDuration = Duration(config.pushNotificationConfig.lockRefreshDurationInMilliseconds, TimeUnit.MILLISECONDS)
+
+  //TODO: should we pass in HeaderCarrier as an implicit parameter for logging?
+  override def processNotificationsFor(csid: ClientSubscriptionId, lockOwnerId: LockOwnerId): Future[Unit] = {
+    //implicit HeaderCarrier required for ApiSubscriptionFieldsConnector
+    //however looking at api-subscription-fields service I do not think it is required so keep new HeaderCarrier() for now
+    implicit val hc = HeaderCarrier()
```

Comment:
it should receive the expiry time from the dispatcher which locked this record. The refresher should refresh it when that time runs out and from that point onwards, it should be running after refreshDuration. The hc can be defined on top of this class and can be used wherever it is required.